common/src/main/java/com/microsoft/identity/common/crypto/wrappedsecretkey/WrappedSecretKeyJsonObjectSerializer.kt

```diff
@@ -0,0 +1,171 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.crypto.wrappedsecretkey
+
+import com.microsoft.identity.common.crypto.wrappedsecretkey.WrappedSecretKeySerializerManager.HEADER_ID_FIELD_SIZE_BYTES
+import com.microsoft.identity.common.crypto.wrappedsecretkey.WrappedSecretKeySerializerManager.METADATA_LENGTH_FIELD_SIZE_BYTES
+import com.microsoft.identity.common.java.exception.ClientException
+import com.microsoft.identity.common.logging.Logger
+import org.json.JSONObject
+import java.nio.ByteBuffer
+
+/**
+ * JSON-based serializer for [WrappedSecretKey] that stores metadata in JSON format.
+ *
+ * This serializer implements version 1 of the wrapped secret key format, storing metadata
+ * as JSON alongside the raw wrapped key data. This format provides better structure and
+ * extensibility compared to the legacy format while maintaining human-readable metadata.
+ *
+ * **Format structure:**
+ * ```
+ * [Header ID: 4 bytes][Metadata Length: 4 bytes][JSON Metadata][Raw Key Data]
+ * ```
+ *
+ * **JSON Metadata fields:**
+ * - `algorithm`: The key algorithm (e.g., "AES")
+ * - `cipherTransformation`: The cipher transformation used (e.g., "RSA/ECB/PKCS1Padding")
+ * - `version`: Format version (1 for this serializer)
+ * - `keyDataLength`: Length of the raw key data for validation
+ *
+ * **Format characteristics:**
+ * - Version: 1
+ * - Header identifier: 0x00FF3C01
+ * - Metadata encoding: UTF-8 JSON
+ * - Supports backward compatibility validation
+ * - Human-readable metadata for debugging
+ *
+ * This serializer bridges the gap between the legacy format (version 0) and more
+ * advanced binary formats, providing structured metadata while maintaining readability.
+ *
+ * @see IWrappedSecretKeySerializer
+ * @see WrappedSecretKey
+ * @see WrappedSecretKeyLegacySerializer
+ */
+class WrappedSecretKeyJsonObjectSerializer: IWrappedSecretKeySerializer {
+
+    companion object {
+        private const val TAG = "WrappedSecretKeyJsonObjectSerializer"
+
+        /** Version identifier for the JSON-based format */
+        const val VERSION = 1
+
+        /** Header identifier used to recognize this format */
+        private const val NEW_FORMAT_HEADER_IDENTIFIER = 0x00FF3C01
```

Comment:
>NEW_FORMAT_HEADER_IDENTIFIER [](http://example.com/codeflow?start=26&length=28)

what is significance of the header?

Reply:
Reworked as discussed offline, the header will only represent a unique identifier to indicate the byte array contains metadata